### PR TITLE
GO-101 Do not allow creating Message with duplicate UUID in Box

### PR DESCRIPTION
**Pull request by luciajanikova**

### CODE REVIEW

File: app/models/upvs/message_draft.rb [hunk 1]
```diff
@@ -66,14 +66,6 @@ def self.load_from_params(message_params, box:)
...
   private
 
-  def validate_uuid
-    if uuid
-      errors.add(:metadata, 'Message ID must be in UUID format') unless uuid.match?(Utils::UUID_PATTERN)
-    else
-      errors.add(:metadata, "Message ID can't be blank")
-    end
-  end
-
```
Comment (luciajanikova) [inline app/models/upvs/message_draft.rb:76]: Toto je duplicita, rovnaka metoda je aj v parent classe, vyhodila som.

File: test/integration/upvs_message_drafts_api_test.rb [hunk 2]
```diff
@@ -120,6 +121,45 @@ class UpvsMessageDraftsApiTest < ActionDispatch::IntegrationTest
...
+  <subject>Všeobecný predmet</subject>
+  <text>Všeobecný text</text>
+</GeneralAgenda>')
+        }
+      ]
+    }
+
+    post '/api/messages/message_drafts', params: message_params.merge({ token: generate_api_token(sub: @tenant.id, key_pair: @key_pair) }), as: :json
+
+    assert_response :unprocessable_entity
```
Comment (jsuchal) [inline test/integration/upvs_message_drafts_api_test.rb:155]: https://stackoverflow.com/questions/3825990/http-response-code-for-post-when-resource-already-exists

**Review by jsuchal — CHANGES_REQUESTED** [review]
dajme iny error code a mozno aj do swaggera napisme

File: app/controllers/api/messages_controller.rb [hunk 3]
```diff
@@ -38,13 +38,18 @@ def message_drafts
         @message.add_cascading_tag(tag)
       end
 
+      unless @message.valid?(:validate_uuid_uniqueness)
+        @message.destroy
```
Comment (jsuchal) [inline app/controllers/api/messages_controller.rb:42]: toto je divne, ak to nie je validne tak by sa to ani nemalo vyrobit ci?

Comment (luciajanikova) [reply app/controllers/api/messages_controller.rb]: Hej, presunula som vyssie pred save

**Review by jsuchal — APPROVED** [review]

**Pull request merged**